pkg/ceph/osd/device.go

```diff
@@ -305,6 +288,71 @@ func partitionOSD(context *clusterd.Context, config *osdConfig) error {
 	return nil
 }
 
+func prepareFilestoreDevice(context *clusterd.Context, config *osdConfig, doFormat bool) error {
+	if !isFilestoreDevice(config) {
+		// nothing to do
+		return nil
```

Comment:
no, there are no valid cases for this to be called when it's not a filestore device.  i'll update it to return an error.